app/src/main/java/com/arctouch/codechallenge/movie_details/MovieDetailsActivity.java

```diff
@@ -0,0 +1,181 @@
+import android.support.v7.widget.Toolbar;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.di.InjectionMovieDetailsPresenter;
+import com.arctouch.codechallenge.extensions.DateExtensionsKt;
+import com.arctouch.codechallenge.extensions.GlideExtensionsKt;
+import com.arctouch.codechallenge.listener.ISnackBarActionListener;
+import com.arctouch.codechallenge.model.Genre;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.movie_details.presentation.MovieDetailsContract;
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder;
+import com.arctouch.codechallenge.util.NetworkUtils;
+import com.arctouch.codechallenge.util.ProgressDialogUtils;
+import com.arctouch.codechallenge.util.SnackUtils;
+
+import java.util.Locale;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class MovieDetailsActivity extends AppCompatActivity implements MovieDetailsContract.View, ISnackBarActionListener {
+
+    @BindView(R.id.coordinator_details)
+    CoordinatorLayout coordinatorLayout;
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.img_context)
+    ImageView imgContext;
+
+    @BindView(R.id.img_poster)
+    ImageView imgPoster;
+
+    @BindView(R.id.img_adult)
+    ImageView imgAdult;
+
+    @BindView(R.id.tv_vote_average)
+    TextView tvVoteAverage;
+
+    @BindView(R.id.tv_genres)
+    TextView tvGenres;
+
+    @BindView(R.id.tv_movie_title)
+    TextView tvMovieTitle;
+
+    @BindView(R.id.tv_release_date)
+    TextView tvReleaseDate;
+
+    @BindView(R.id.tv_overview)
+    TextView tvOverview;
+
+    private ProgressDialog progress;
+    private MovieDetailsContract.Presenter movieDetailsPresenter;
+    private Bundle args;
+    private Movie movie;
+    private MovieImageUrlBuilder movieImageUrlBuilder;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_movie_details);
+
+        ButterKnife.bind(this);
+
+        setSupportActionBar(toolbar);
+        ActionBar ab = getSupportActionBar();
+        if (ab != null) {
+            ab.setTitle(R.string.app_name);
+            ab.setDisplayHomeAsUpEnabled(true);
+            ab.setDisplayShowHomeEnabled(true);
+        }
+
+        args = getIntent().getExtras();
+
+        movieImageUrlBuilder = new MovieImageUrlBuilder();
+
+        movieDetailsPresenter = InjectionMovieDetailsPresenter.inject(this, this, Schedulers.io(), AndroidSchedulers.mainThread());
+
+        if(args != null && args.getParcelable(Movie.BUNDLE) == null) {
+            if (savedInstanceState != null && savedInstanceState.getParcelable(Movie.BUNDLE) != null) {
```

Comment:
Can you explain what `savedInstanceState` is and why it is necessary?

Reply:
savedInstanceState is a Bundle, which is the state of the app in which was saved before the app’s activity was destroyed. It is necessary to retrieve data from the previous screen state, like the Movie. One thing to consider is that this Bundle savedInstaceState has size limitations in Android Nougat (7.0), which is around 500kb, so if there was an image in this Bundle greater than 500kb, the app would crash. One solution for it would be storing this image in a temporary SharedPreferences (using repository, etc) so, when retrieving this image after rotating the screen, it could be deleted.